src/test/java/bio/overture/ego/controller/PolicyControllerTest.java

```diff
@@ -0,0 +1,295 @@
+  @SneakyThrows
+  public void addUserPermission() {
+    val policyId = entityGenerator.setupSinglePolicy("AddUserPermission").getId().toString();
+    val userId = entityGenerator.setupUser("UserPolicy Add").getId().toString();
+
+    val entity = new HttpEntity<String>("READ", headers);
+    val response =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/permission/user/%s", policyId, userId)),
+            HttpMethod.POST,
+            entity,
+            String.class);
+
+    val responseStatus = response.getStatusCode();
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    // TODO: Fix it so that POST returns JSON, not just random string message
+
+    val getResponse =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/users", policyId)),
+            HttpMethod.GET,
+            new HttpEntity<String>(null, headers),
+            String.class);
+
+    val getResponseStatus = getResponse.getStatusCode();
+    val getResponseJson = MAPPER.readTree(getResponse.getBody());
+    val groupPermissionJson = getResponseJson.get(0);
+
+    assertThat(getResponseStatus).isEqualTo(HttpStatus.OK);
+    assertThat(groupPermissionJson.get("id").asText()).isEqualTo(userId);
+    assertThat(groupPermissionJson.get("mask").asText()).isEqualTo("READ");
+  }
+
+  @Test
+  @SneakyThrows
+  public void deleteUserPermission() {
+    val policyId = entityGenerator.setupSinglePolicy("DeleteGroupPermission").getId().toString();
+    val userId = entityGenerator.setupUser("UserPolicy Delete").getId().toString();
+
+    val entity = new HttpEntity<String>("WRITE", headers);
+    val response =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/permission/user/%s", policyId, userId)),
+            HttpMethod.POST,
+            entity,
+            String.class);
+
+    val responseStatus = response.getStatusCode();
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    // TODO: Fix it so that POST returns JSON, not just random string message
+
+    val deleteResponse =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/permission/user/%s", policyId, userId)),
+            HttpMethod.DELETE,
+            new HttpEntity<String>(null, headers),
+            String.class);
+
+    val deleteResponseStatus = deleteResponse.getStatusCode();
+    assertThat(deleteResponseStatus).isEqualTo(HttpStatus.OK);
+
+    val getResponse =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s/users", policyId)),
+            HttpMethod.GET,
+            new HttpEntity<String>(null, headers),
+            String.class);
+
+    val getResponseStatus = getResponse.getStatusCode();
+    val getResponseJson = (ArrayNode) MAPPER.readTree(getResponse.getBody());
+
+    assertThat(getResponseStatus).isEqualTo(HttpStatus.OK);
+    assertThat(getResponseJson.size()).isEqualTo(0);
```

Comment:
create another request to ensure the user nor the policy was deleted in the `/policies/%s/users` request. Like the other delete operation above, this method should only delete the relationship